src/mesh/meshblock.hpp

```diff
@@ -182,14 +182,15 @@ class MeshBlock : public std::enable_shared_from_this<MeshBlock> {
     par_dispatch_(std::forward<Args>(args)...);
   }
 
-  // 5D default loop pattern
   template <typename Function>
-  inline void par_for(const std::string &name, const int &bl, const int &bu,
-                      const int &nl, const int &nu, const int &kl, const int &ku,
-                      const int &jl, const int &ju, const int &il, const int &iu,
-                      const Function &function) {
-    parthenon::par_for(DEFAULT_LOOP_PATTERN, name, exec_space, bl, bu, nl, nu, kl, ku, jl,
-                       ju, il, iu, function);
+  inline void par_for_bndry(const std::string &name, const IndexRange &nb,
+                            const IndexDomain &domain, const bool coarse,
+                            const Function &function) {
+    auto bounds = coarse ? c_cellbounds : cellbounds;
+    auto ib = bounds.GetBoundsI(domain);
+    auto jb = bounds.GetBoundsJ(domain);
+    auto kb = bounds.GetBoundsK(domain);
```

Comment:
const?

Reply:
Wait will that actually work? These functions return a new, initialized object. I think the move operator is used.
```C++
  KOKKOS_INLINE_FUNCTION const IndexRange GetBoundsI(const IndexDomain &domain) const
      noexcept {
    return (domain == IndexDomain::interior) ? x_[0] : IndexRange{is(domain), ie(domain)};
  }

  KOKKOS_INLINE_FUNCTION const IndexRange GetBoundsJ(const IndexDomain &domain) const
      noexcept {
    return (domain == IndexDomain::interior) ? x_[1] : IndexRange{js(domain), je(domain)};
  }

  KOKKOS_INLINE_FUNCTION const IndexRange GetBoundsK(const IndexDomain &domain) const
      noexcept {
    return (domain == IndexDomain::interior) ? x_[2] : IndexRange{ks(domain), ke(domain)};
  }
```

Reply:
Ya your right, that's my bad.